make/autoconf/jdk-options.m4

```diff
@@ -445,6 +445,46 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
   AC_SUBST(ASAN_ENABLED)
 ])
 
+###############################################################################
+#
+# UndefinedBehaviorSanitizer
```

Comment:
I think this logic fits better in `flags.m4`, otherwise this looks ok to me.

Reply:
Ah now I understand that this compiles runtime checks into the product. In that case it does actually fit well into jdk-options.m4, so you can leave it there.

Reply:
Well, this function actually does two things -- first it checks if ubsan support should be enabled, and that code really belong here. But then if it i enabled, it adds additional compiler options, including globally silencing some warnings, and that should really reside in flags-cflags.m4. 

But, I see we have this pattern for other optional tooling support like ASan and GCov, so I guess this is fine for now.

I'm really looking forward to a better way of handling flags so we wouldn't have to do it like this.